charms/knative-operator/src/charm.py

```diff
@@ -135,18 +198,39 @@ def _main(self, event):
 
         # Apply Kubernetes resources
         self.unit.status = MaintenanceStatus("Applying resources")
-        self._apply_all_resources()
+        self._apply_resources(resource_handler=self.resource_handler)
```

Comment:
oh I think I understand, is this because it is used for both regular and observability resource_handler objects?  If so, that makes sense to me and you can resolve this

Reply:
> oh I think I understand, is this because it is used for both regular and observability resource_handler objects? If so, that makes sense to me and you can resolve this

Yes